### PR TITLE
[7851] Show the performance banner

### PR DESCRIPTION
**Pull request by defong**
### Context
performance banner

### Changes proposed in this pull request
Show the performance banner

### Guidance to review
Builds off from this PR https://github.com/DFE-Digital/register-trainee-teachers/pull/4828

As an accredited provider
image

As an lead partner
image

The `Sign off your performance profile` link is out scope as the page is not built yet (next cards etc).

### Important business

- [ ] Does this PR introduce any PII fields that need to be overwritten or deleted in db/scripts/sanitise.sql?
- [ ] Does this PR change the database schema? If so, have you updated the config/analytics.yml file and considered whether you need to send 'import_entity' events?
- [ ] Do we need to send any updates to DQT as part of the work in this PR?
- [ ] Does this PR need an ADR?

NB: Please notify the #twd_data_insights team and ask for a review if new fields are being added to analytics.yml

### CODE REVIEW

**Comment by defong** [comment]
[Performance banner](https://trello.com/c/IbVHD79j/7851-performance-banner)

File: app/components/performance_profile_banner/view.rb [hunk 1]
```diff
@@ -18,10 +18,10 @@ def banner_heading_text
       "The #{previous_academic_cycle_label} ITT performance profile sign off due"
     end
 
-    delegate :label, :end_year, to: :previous_academic_cycle, prefix: true
+    delegate :label, :end_date_of_performance_profile, to: :previous_academic_cycle, prefix: true
 
     def deadline_date
-      "28 February #{previous_academic_cycle_end_year + 1}"
+      previous_academic_cycle.end_date_of_performance_profile.strftime("%-d %B %Y")
```
Comment (Reyko) [inline app/components/performance_profile_banner/view.rb:24]:
```suggestion
      previous_academic_cycle.end_date_of_performance_profile.strftime(Date::DATE_FORMATS[:govuk])
```

File: spec/features/performance_profile_banner_spec.rb [hunk 2]
```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+feature "performance profile banner" do
+  context "within the performance profile date range" do
+    background {
+      Timecop.freeze(AcademicCycle.previous.performance_profile_date_range.to_a.sample)
+    }
```
Comment (Reyko) [inline spec/features/performance_profile_banner_spec.rb:9]:
```suggestion
    background do
      Timecop.freeze(AcademicCycle.previous.performance_profile_date_range.to_a.sample)
    end
```

**Review by Reyko — APPROVED** [review]

**Comment by andrewscrivener** [comment]
@defong Looks good, but you've missed the word **is**

Should be
The 2023-2024 ITT performance profile sign off **is** due

**Comment by andrewscrivener** [comment]
:shipit:

**Pull request merged**